src/Symfony/Component/Validator/Tests/Validator/AbstractValidatorTest.php

```diff
@@ -1081,6 +1084,50 @@ public function testValidateMultipleGroups()
         $this->assertCount(2, $violations);
     }
 
+    public function testValidateMultipleGroupsForCollectionConstraint()
```

Comment:
This test belongs in CollectionValidatorTest. Also, I'm missing tests for what is happening when one or more groups are set on the `Collection` constraint itself. You should systematically test the different permutations:

* `Collection` group equals that of the constraints
* `Collection` group does not equal that of the constraints
* `Collection` groups are a superset of the groups of the constraints
* `Collection` groups are a subset of the groups of the constraints
* `Collection` groups and constraing groups don't overlap

The same applies to AllValidator.